db/view/view.cc

```diff
@@ -2563,7 +2563,24 @@ future<> view_builder::do_build_step() {
     });
 }
 
-future<> view_builder::migrate_to_v2(db::system_keyspace& sys_ks, cql3::query_processor& qp, service::raft_group0_client& group0_client, abort_source& as, service::group0_guard guard) {
+future<> view_builder::generate_mutations_on_node_left(replica::database& db, cql3::query_processor& qp, api::timestamp_type timestamp, locator::host_id host_id, std::vector<canonical_mutation>& muts) {
+    const sstring query_string = format("DELETE FROM {}.{} WHERE keyspace_name = ? AND view_name = ? AND host_id = ?",
+            db::system_keyspace::NAME, db::system_keyspace::VIEW_BUILD_STATUS_V2);
+
+    for (auto& view : db.get_views()) {
+        // TODO MICHAEL only views for which this node was a replica ?
+        auto vb_muts = co_await qp.get_mutations_internal(
+                query_string,
+                view_builder_query_state(),
+                timestamp,
+                {view->ks_name(), view->cf_name(), host_id.uuid()});
+        assert(vb_muts.size() == 1);
+        muts.push_back(canonical_mutation(std::move(vb_muts[0])));
+        co_await coroutine::maybe_yield();
+    }
```

Comment:
Currently when a node is removed I delete its rows by going over all the existing views and generating a mutation for each view. This is probably not the most efficient because there could be many views for which this node is not a replica, and these rows will not exist.
I'm not sure if there is a better way to do this. Maybe first find all the node's rows with SELECT? or is there a way to get only the views for which this node was a replica?

Reply:
on second thought, I think every node has a row for each view, doesn't matter if it's a replica or not, so it's ok to generate a mutation for each view.